wdl-lint/src/rules/disallowed_declaration_name.rs

```diff
@@ -0,0 +1,295 @@
+    }
+
+    fn description(&self) -> &'static str {
+        "Declaration names should not end with their type"
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Declaration names should not include their type as a suffix. \
+        This makes the code more verbose and often redundant. For example, use \
+        'counter' instead of 'counterInt', or 'is_active' instead of 'activeBool'."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Style, Tag::Clarity])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [SyntaxKind]> {
+        Some(&[
+            SyntaxKind::VersionStatementNode,
+            SyntaxKind::BoundDeclNode,
+            SyntaxKind::UnboundDeclNode,
+            SyntaxKind::InputSectionNode,
+            SyntaxKind::OutputSectionNode,
+        ])
+    }
+}
+
+/// Check declaration name for type suffixes only
+fn check_decl_name(
+    state: &mut Diagnostics,
+    decl: &Decl,
+    exceptable_nodes: &Option<&'static [SyntaxKind]>,
+) {
+    let name = decl.name();
+    let name_str = name.as_str();
+    let name_lower = name_str.to_lowercase();
+    
+    // Get the declaration type
+    let ty = decl.ty();
+    
+    // Skip type reference types (user-defined structs)
+    if ty.as_type_ref().is_some() {
+        return;
+    }
+    
+    // Extract type names to check
+    let mut type_names = Vec::new();
+    
+    // Add primitive type if present
+    if let Some(primitive_type) = ty.as_primitive_type() {
+        // Skip File and String types as they cause too many false positives
+        match primitive_type.kind() {
+            PrimitiveTypeKind::File | PrimitiveTypeKind::String => {},
+            PrimitiveTypeKind::Boolean => {
+                // For Boolean, check both "Boolean" and "Bool"
+                type_names.push("Boolean".to_string());
+                type_names.push("Bool".to_string());
+            },
+            _ => type_names.push(primitive_type.to_string()),
+        }
+    }
+    
+    // Add compound types
+    if ty.as_array_type().is_some() {
+        type_names.push("Array".to_string());
+    } else if ty.as_map_type().is_some() {
+        type_names.push("Map".to_string());
+    } else if ty.as_pair_type().is_some() {
+        type_names.push("Pair".to_string());
+    }
+    
+    // Check each type name against the declaration name (only as a suffix)
+    for type_name in &type_names {
+        let type_lower = type_name.to_lowercase();
+        
+        // Skip if the type name is too short
+        if type_lower.len() < 3 {
+            continue;
+        }
+        
+        // Special handling for Int - check only as a full suffix
+        if type_lower == "int" {
+            // Only check if it appears as a complete suffix
+            if name_lower.ends_with(&type_lower) {
+                // Skip common false positive suffixes
+                let common_false_positives = ["point", "print", "hint", "lint", "mint", "tint", "stint"];
```

Comment:
This hardcoded approach seems too fragile and difficult to maintain to me.
We already depend on the `convert_case` crate, which has some useful capabilities. I probably wouldn't want to add that dependency just for this case, but we're already compiling it! So might as well use it.

Check out the `convert_case` crate, and see if we can use it here. My thinking is that we can split the Decl name on common word boundaries, and see if `int` appears on it's own (i.e. not as part of a longer word like `point`).